### ISSUE TITLE
would be nice if inflictrandomstatus allowed modders to add their own statuses

### ISSUE THREAD

**Issue by ncrecc** [TerryCavanagh/diceydungeons.com#2628]
i.e. instead of
```haxe
if(_target.isplayer){
  //If we're targeting the player we can consider some extra stuff
  randstatus = rand(["curse", "lock", "fire", "ice", "weaken", "shock", "poison", "blind", "haunted"]);
}else{
  randstatus = rand(["curse", "lock", "fire", "ice", "weaken", "shock", "poison"]);
}
```
do
```haxe
if(_target.isplayer){
  //If we're targeting the player we can consider some extra stuff
  randstatus = rand(loadtext("reunion/randomstatuses_onplayer"));
}else{
  randstatus = rand(loadtext("reunion/randomstatuses_onenemy"));
}
```

**Comment by JuniorUdale** [comment]
Also as an aside, it'd be really nice if this was added to the base game and used in items there (e.g. Dark Shadow). It'd make it easier for modders to include their own "inflict a random status" items (that also inflict modded statuses) if there was a base game script for it.

**Comment by TerryCavanagh** [comment]
Whoops, I missed this one while writing up issues! I'll close the other one 😅

**Comment by TerryCavanagh** [comment]
> Also as an aside, it'd be really nice if this was added to the base game and used in items there (e.g. Dark Shadow). It'd make it easier for modders to include their own "inflict a random status" items (that also inflict modded statuses) if there was a base game script for it.

Yep, that makes a lot of sense! The best solution to this might be to add this script to vanilla, and have Reunion append Haunted/Blind etc to the lists so that vanilla doesn't change, but the new behaviour is supported.

There's the slight complication of sound effects not always matching the status names (e.g. Haunted uses the `_curse` sound event), but that's fixable by reading in a csv instead of a single table.
